src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs

```diff
@@ -99,6 +99,33 @@ public bool IsAbsoluteTemplate
             };
         }
 
+        /// <summary>
+        /// Combines the prefix and route template for an attribute route.
+        /// </summary>
+        /// <param name="prefix">The prefix.</param>
+        /// <param name="template">The route template.</param>
+        /// <returns>The combined pattern.</returns>
+        public static string CombineTemplates(string prefix, string template)
+        {
+            var result = CombineCore(prefix, template);
+            return CleanTemplate(result);
+        }
+
+        /// <summary>
+        /// Determines if a template pattern can be used to override a prefix.
+        /// </summary>
+        /// <param name="template">The template.</param>
+        /// <returns><c>true</c> if this is an overriding template, <c>false</c> otherwise.</returns>
+        /// <remarks>
+        /// Route templates starting with "~/" or "/" can be used to override the prefix.
+        /// </remarks>
+        public static bool IsOverridePattern(string template)
```

Comment:
Similar reason as above, see comment below. This is tied to Attribute routing and we shouldn't reuse it for razor pages.